### DISCUSSION TITLE
NEW API (has && get) in new Map()

### DISCUSSION THREAD

**Discussion by jojojojojoj5564656465465** [open-circle/valibot#895] [Ideas]
![image](https://github.com/user-attachments/assets/9a9c0e53-d63e-4b35-935d-59eb0bf4a04c)

**Comment by fabian-hiller** [comment]
Great idea! I would probably name these actions `hasItem` and `getItem`, so they match with other actions like `mapItems` and `findItem`. What do you think?

```ts
import * as v from 'valibot';

const Schema = v.pipe(
  v.map(v.string(), v.number()),
  v.hasItem('foo', 'Custom error message!')
);
```

**Reply by jojojojojoj5564656465465** [reply]
Hi,

I would like to share some improvement suggestions for valibot:

1. Regarding API naming:
- I suggest simplifying to more explicit names like `map_get`, `array_filter`, `array_split`
- `map_has` could be merged with `map_get` since the functionality would be redundant

2. Community aspect:
- It would be valuable to have a dedicated community page
- This would encourage third-party extension development compatible with valibot
- Example: different developers could create their own parsers (like `s.parse()` or `e.parse()`) while maintaining compatibility with the valibot ecosystem

The goal is to keep the API simple and intuitive while fostering the growth of an active community around the project.

What are your thoughts on this?

Best regards

**Reply by fabian-hiller** [reply]
> I suggest simplifying to more explicit names like `map_get`, `array_filter`, `array_split`

The idea is good, but there are some drawbacks. Something like `map_has` would use exactly the same implementation as `set_has`. So it makes more sense to call it `hasItem` and make it compatible with maps and sets. That's also how our other actions like `minLength` work. Instead of `string_min` and `array_min` there is just `minLength` which can be used for both.

> It would be valuable to have a dedicated community page

Do you mean something like our ecosystem page? Or do you mean a special guide for library authors that explains how to create libraries on top of Valibot? We also have a Discord community server.

https://valibot.dev/guides/ecosystem/

**Reply by jojojojojoj5564656465465** [reply]
hasItem could works with object ? 

const check = (x:string): => x in obj


for map//
  v.custom<keyof typeof size>((input) =>
    !!Object.keys(size).find((e) => e === input)
  ),

**Reply by fabian-hiller** [reply]
Probably not, it will only work with objects that contain a `.has(...)` method like a map or a set.
